pkg/operator/highcpuusagesnoalertcontroller/highcpuusagesnoalert_controller_test.go

```diff
@@ -0,0 +1,90 @@
+package highcpuusagesnoalertcontroller
+
+import (
+	"bytes"
+	"context"
+	"reflect"
+	"strconv"
+	"testing"
+
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/bindata"
+	"k8s.io/client-go/dynamic"
+)
+
+func Test_snoAlert(t *testing.T) {
+	type args struct {
+		ctx                 context.Context
+		client              dynamic.Interface
+		enabledCapabilities []v1.ClusterVersionCapability
+		cpuMode             v1.CPUPartitioningMode
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "No node tuning capability",
+			args: args{
+				ctx: context.TODO(),
+			},
+			want:    alertWithGivenCores(8),
+			wantErr: false,
+		},
+		{
+			name: "Node tuning capability, but wrong cpu mode",
+			args: args{
+				ctx:                 context.TODO(),
+				enabledCapabilities: []v1.ClusterVersionCapability{v1.ClusterVersionCapabilityNodeTuning},
+				cpuMode:             v1.CPUPartitioningNone,
+			},
+			want:    alertWithGivenCores(8),
+			wantErr: false,
+		},
+		{
+			name: "Node tuning capability,correct cpu mode, but no PerformanceProfile",
+			args: args{
+				ctx:                 context.TODO(),
+				enabledCapabilities: []v1.ClusterVersionCapability{v1.ClusterVersionCapabilityNodeTuning},
+				cpuMode:             v1.CPUPartitioningAllNodes,
+				client:              clientWithObjects(),
+			},
+			want:    alertWithGivenCores(8),
+			wantErr: false,
+		},
+		{
+			name: "Node tuning capability, correct cpu mode, correct PerformanceProfile",
+			args: args{
+				ctx:                 context.TODO(),
+				enabledCapabilities: []v1.ClusterVersionCapability{v1.ClusterVersionCapabilityNodeTuning},
+				cpuMode:             v1.CPUPartitioningAllNodes,
+				client:              clientWithObjects(performanceProfileWithNodeSelector("node-role.kubernetes.io/master")),
+			},
+			want:    alertWithGivenCores(2),
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := snoAlert(tt.args.ctx, tt.args.client, tt.args.enabledCapabilities, tt.args.cpuMode)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("snoAlert() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("snoAlert() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func alertWithGivenCores(cores int) []byte {
+	fileData, err := bindata.Asset("assets/alerts/cpu-utilization-sno.yaml")
```

Comment:
we usually compare against pre-computed files, for example https://github.com/openshift/cluster-authentication-operator/blob/master/pkg/operator/workload/sync_openshift_oauth_apiserver_test.go#L71